MapboxNavigation/ErrorCodes.swift

```diff
@@ -0,0 +1,36 @@
+import Foundation
+
+/**
+ Enum used for indicating the type of error occured while speaking an instruction.
+ */
+public enum SpokenVoiceError: Int {
+    /**
+     Default error.
+     */
+    case defaultError
+    
+    /**
+     An intruction interrupted another instruction.
+     */
+    case overlappingInstruction
+    
+    /**
+     The audio player failed to preprare to play audio data.
+     */
+    case audioPlayerFailedToPrepare
```

Comment:
Some of these cases are specific to PollyVoiceController (or at least are inapplicable to RouteVoiceController). Perhaps we can generalize these cases so they can be used for more kinds of voice controllers in the future.